python/ray/autoscaler/_private/commands.py

```diff
@@ -923,6 +923,13 @@ def _set_up_config_for_head_node(
     # head node won't be able to connect to workers
     remote_config["auth"].pop("ssh_proxy_command", None)
 
+    # Drop the head_node field if it was introduced. It is technically not a
+    # valid field in the config, but it may have been introduced after
+    # validation (see _bootstrap_config() call to
+    # provider_cls.bootstrap_config(config)). The head node will never try to
+    # launch a head node so it doesn't need these defaults.
+    remote_config.pop("head_node", None)
```

Comment:
nice going. that's sharp thinking for past 11pm. 
this is definitely a hack to undo some hacks, but we'll take it

Reply:
agreed it's not perfect, but feels manageable to throw all the related hacks in one place here.